pilot/pkg/model/push_context.go

```diff
@@ -1612,7 +1612,23 @@ func (ps *PushContext) initEnvoyFilters(env *Environment) error {
 		return err
 	}
 
-	sortConfigByCreationTime(envoyFilterConfigs)
+	sort.SliceStable(envoyFilterConfigs, func(i, j int) bool {
+		ifilter := envoyFilterConfigs[i].Spec.(*networking.EnvoyFilter)
+		jfilter := envoyFilterConfigs[j].Spec.(*networking.EnvoyFilter)
+		if ifilter.Priority != jfilter.Priority {
+			return ifilter.Priority < jfilter.Priority
+		}
+		// If prirority is same fallback to name and creation timestamp, else use prirority.
+		// If creation time is the same, then behavior is nondeterministic. In this case, we can
+		// pick an arbitrary but consistent ordering based on name and namespace, which is unique.
+		// CreationTimestamp is stored in seconds, so this is not uncommon.
+		if envoyFilterConfigs[i].CreationTimestamp == envoyFilterConfigs[j].CreationTimestamp {
```

Comment:
@ramaraochavali the same pattern can be applied in the 2nd comparison.

if a != b return a < b

it does not change the nesting now, but just makes it consistent.
Right now the condition is on timestamp but block logic is on name.